### PR TITLE
Incorporate S3 object and bucket IDs (GSI 217)

### PR DESCRIPTION
**Pull request by TheByronHimes**
Adds an object ID to the FileMetadata model after generating it in the UploadService while initiating a new upload. The object is assigned at the same time as the latest_upload_id field. UploadAttempt model now also features the object_id field.

### CODE REVIEW

File: ucs/core/file_service.py [hunk 1]
```diff
@@ -65,7 +65,9 @@ def _assert_update_allowed(
     async def _insert_new(self, file: models.FileMetadataUpsert) -> None:
         """Create a metadata entry for a new file."""
 
-        full_metadata = models.FileMetadata(**file.dict(), latest_upload_id=None)
+        full_metadata = models.FileMetadata(
+            **file.dict(), latest_upload_id=None, object_id=None
```
Comment (KerstenBreuer) [inline ucs/core/file_service.py]: The object ID is only relevant in the context of an upload attempt.

File: ucs/core/models.py [hunk 2]
```diff
@@ -100,6 +103,9 @@ class FileMetadata(FileMetadataUpsert):
     A model containing the full metadata on a file.
     """
 
+    object_id: Optional[str] = Field(
+        None, description="Object ID assigned during latest upload attempt."
+    )
```
Comment (KerstenBreuer) [inline ucs/core/models.py]: No need to duplicate the information

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]
Great

File: ucs/core/file_service.py (hunk 1 above)
Comment (TheByronHimes) [reply ucs/core/file_service.py]: Yeah, you're right. The few places the ID was extracted from the metadata object can just have the id supplied as a parameter. See fix in [09eae94](https://github.com/ghga-de/upload-controller-service/pull/45/commits/09eae94e0b6433c962662c63f2b0f3bad65d9b8e) and [f4df54f](https://github.com/ghga-de/upload-controller-service/pull/45/commits/f4df54fc15606333ddb4e33ef0512250d2a04081)

File: ucs/core/models.py (hunk 2 above)
Comment (TheByronHimes) [reply ucs/core/models.py]: fixed

File: ucs/core/file_service.py (hunk 1 above)
Comment (KerstenBreuer) [inline ucs/core/file_service.py]:
```suggestion
            **file.dict(), latest_upload_id=None
```

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]

**Review by mephenor — APPROVED** [review]

File: ucs/core/file_service.py (hunk 1 above)
Comment (TheByronHimes) [reply ucs/core/file_service.py]: Gah, fixed it.

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**